### PR TITLE
Fix slack GitHub action

### PR DESCRIPTION
**Pull request by zaneselvans**
# Overview

Update our usage of `slack-github-action` to be compatible with the new v2.0 of that action.

# Testing

The same action failed in the main PUDL repo and this fix worked there. Just updating the syntax here. Have not actually tested (but the old way would definitely fail).

### CODE REVIEW

**Review by e-belfer — CHANGES_REQUESTED** [review]
Running the `early_final_release_checker` action [returned something](https://catalystcooperative.slack.com/archives/C03FHB9N0PQ/p1732112369041809) that's clearly not formatted correctly.

> {"attachments":[{"blocks":[{"type":"header","text":{"type":"plain_text","text":"Archiver Run Outcomes"}},{"type":"section","text":{"type":"mrkdwn","text":"Unchanged"}},{"type":"section","text":{"type":"mrkdwn","text":"[eia860](https://zenodo.org/deposit/14192827)\nNo changes."}},{"type":"section","text":{"type":"mrkdwn","text":"[eia861](https://zenodo.org/deposit/14192826)\nNo changes."}},{"type":"section","text":{"type":"mrkdwn","text":"[eia923](https://zenodo.org/deposit/14192828)\nNo changes."}}]}]}

**Comment by zaneselvans** [comment]
Okay I think this is working correctly now.  I switched to using the `blocks` input to the [`chat.postMessage()` method](https://api.slack.com/methods/chat.postMessage) and modified the script to generate an array of blocks.

File: scripts/make_slack_notification_message.py [hunk 1]
```diff
@@ -122,16 +122,10 @@ def section_block(text: str) -> dict:
...
-                        + failed_blocks
-                        + changed_blocks
-                        + unchanged_blocks,
-                    }
-                ]
-            },
+            [header_block("Archiver Run Outcomes")]
+            + failed_blocks
+            + changed_blocks
+            + unchanged_blocks,
```
Comment (zaneselvans) [inline scripts/make_slack_notification_message.py:128]: An array of blocks, as required for the `blocks` input to the [`chat.postMessage()` method](https://api.slack.com/methods/chat.postMessage).

**Review by e-belfer — APPROVED** [review]
Looks like the slackbot worked as expected this time, thanks for fixing it!

**Pull request merged**